### PR TITLE
Fixes Dataset split with that codename already exists.

### LINKED ISSUE #1294: 'Dataset split with that codename already exists' error
## Current Scenario

When a user creates a challenge using a zip configuration that has some dataset split codename, then the challenge is created successfully. If he tries to create a similar challenge with similar dataset split name then he gets an error that `Dataset split with that codename already exists` which should not happen since different challenges can have same dataset split codename. 

## Deliverables

- [ ] Remove `unique=True` from DatasetSplit's codename field
- [ ] Set unique_together `pk` and `dataset_split` in Challenge phase model
- [ ] Don't forget to name migrations

### PR DESCRIPTION
**Pull request by isht3**
* Remove unique=True from DatasetSplit's codename field
* Set unique_together pk and dataset_split in Challenge phase model
* Add migrations for the changes.

Fixes #1294.

### CODE REVIEW

**Comment by deshraj** [comment]
@isht3 I dont think the change  
- Set unique_together pk and dataset_split in Challenge phase model

is incorporated in this PR.

File: apps/challenges/models.py [hunk 1]
```diff
@@ -124,7 +124,7 @@ def __init__(self, *args, **kwargs):
     class Meta:
         app_label = 'challenges'
         db_table = 'challenge_phase'
-        unique_together = (('codename', 'challenge'),)
+        unique_together = (('codename', 'challenge'), ('id', 'codename'),)
```
Comment (deshraj) [inline apps/challenges/models.py]: We don't need this change.

**Comment by isht3** [comment]
@deshraj That's where I was confused, I tried `('pk', 'codename')` where `pk` is the primary key of the ChallengePhase model (I think), then I got this `challenges.ChallengePhase: (models.E012) 'unique_together' refers to the non-existent field 'pk'.` Then I thought since `pk` and `id` are similar, the changes would be the same.

File: apps/challenges/models.py (hunk 1 above)
Comment (deshraj) [reply apps/challenges/models.py]: So, we need the additional `unique_together` here with attributes `pk` and `dataset_split`

**Comment by deshraj** [comment]
To clear your doubt: in the models, `pk` and `id` are the same. It should work if you are using `pk` instead of `id`. You might be making some mistake somewhere.

**Comment by isht3** [comment]
@deshraj Okay, Ill look into what's wrong.

**Comment by isht3** [comment]
@deshraj It's a ManyToMany field. What we are trying to achieve is to have the unique DatasetSplit name only within a single challenge, yes?

**Comment by isht3** [comment]
@deshraj I don't think it's a error on my side, because the travis tests are also failing on the `pk` field. I think this answer discusses a potential solution https://stackoverflow.com/questions/4870935/django-enforcing-manytomanyfield-unique-items Should I look into it?

**Comment by deshraj** [comment]
@isht3 can you try replacing `pk` -> `id` here (https://github.com/Cloud-CV/EvalAI/pull/1327/files#diff-06139830cb2f143920edd3015bb7b1baR127) and see if that works.

**Comment by isht3** [comment]
@deshraj `ProgrammingError: column "dataset_split" named in key does not exist` was raised, should I look into other solutions?

**Pull request closed without merging**